### PR TITLE
Validate branch names on the settings page

### PR DESCRIPTION
**Pull request by el-mapache**
Screenshot:

screen shot 2018-01-17 at 3 06 48 pm

### CODE REVIEW

File: frontend/components/site/SiteSettings/BasicSiteSettings.jsx [hunk 1]
```diff
@@ -45,15 +46,13 @@ export const BasicSiteSettings = ({
...
-          Branch name:
-        </label>
-        <Field
-          component="input"
+        <BranchField
+          label="Branch name:"
           type="text"
           id="defaultBranchInput"
           name="defaultBranch"
           className="form-control"
```
Comment (jseppi) [inline frontend/components/site/SiteSettings/BasicSiteSettings.jsx]: What do you think about either setting a blank placeholder on the demoBranch field or adding a non-blank one here?

Comment (el-mapache) [reply frontend/components/site/SiteSettings/BasicSiteSettings.jsx]: My feeling here is that its never going to be blank in the above example, I can't really think of any non-broken page state where the site isnt going to have a primary branch. We can duplicate the demoBranch value if it seems safer.

Comment (jseppi) [reply frontend/components/site/SiteSettings/BasicSiteSettings.jsx]: Yeah, that's a fair point. I don't feel strongly about it.

Comment (el-mapache) [reply frontend/components/site/SiteSettings/BasicSiteSettings.jsx]: I ended up changing, it is very simple, like you said, and it doesn't hurt anything 😄

**Comment by wslack** [comment]
woot!

**Pull request merged**